RedacteurPortaal.Tests/Api/ArchiveControllerTests.cs

```diff
@@ -0,0 +1,589 @@
+    //    for (var i = 0; i < updatedArchiveResult?.ContactDetails.Count; i++)
+    //    {
+    //        Assert.IsNotNull(updatedArchiveResult.ContactDetails[i].Email);
+    //        Assert.IsNotNull(updatedArchiveResult.ContactDetails[i].TelephoneNumber);
+    //        Assert.IsNotNull(updatedArchiveResult.ContactDetails[i].Name);
+    //    }
+    //}
+
+    [TestMethod]
+    public async Task CanCreateArchive()
+    {
+        var application = new RedacteurPortaalApplication();
+        var client = application.CreateClient();
+
+        var addArchiveRequest = ArchiveDtoBuilder.BuildAddArchiveRequest();
+        var archiveResult = await client.PostAsJsonAsync("/api/Archive", addArchiveRequest);
+        var resultString = await archiveResult.Content.ReadAsStringAsync();
+
+        var result = JsonSerializer.Deserialize<ArchiveDto>(resultString, new JsonSerializerOptions() { PropertyNameCaseInsensitive = true });
+
+        Assert.IsNotNull(result);
+        Assert.AreEqual(addArchiveRequest.Title, result?.Title);
+        Assert.AreEqual(addArchiveRequest.Label, result?.Label);
+        for (int i = 0; i < addArchiveRequest.MediaPhotoItems.Count; i++)
+        {
+            Assert.AreEqual(addArchiveRequest.MediaPhotoItems[i], result?.MediaPhotoItems?[i]);
+        }
+        for (int i = 0; i < addArchiveRequest.MediaVideoItems.Count; i++)
+        {
+            Assert.AreEqual(addArchiveRequest.MediaVideoItems[i], result?.MediaVideoItems?[i]);
+        }
+        for (int i = 0; i < addArchiveRequest.MediaAudioItems.Count; i++)
+        {
+            Assert.AreEqual(addArchiveRequest.MediaAudioItems[i], result?.MediaAudioItems?[i]);
+        }
+        for (int i = 0; i < addArchiveRequest.NewsItems.Count; i++)
+        {
+            Assert.AreEqual(addArchiveRequest.NewsItems[i], result?.NewsItems?[i]);
+        }
+        //Assert.AreEqual(addArchiveRequest.ArchivedDate, result?.ArchivedDate);
+        CollectionAssert.AreEqual(addArchiveRequest.Scripts, result?.Scripts);
+
+        var newArchive = await client.GetFromJsonAsync<List<ArchiveDto>>("/api/Archive");
+
+        Assert.IsNotNull(newArchive);
+        Assert.IsTrue(newArchive?.Count == 1);
+    }
+
+    [TestMethod]
+    public async Task CanAddVideoItemById()
+    {
+        var application = new RedacteurPortaalApplication();
+        var client = application.CreateClient();
+
+        var addArchiveRequest = ArchiveDtoBuilder.BuildSmallestArchive();
+        var archiveResult = await client.PostAsJsonAsync("/api/Archive", addArchiveRequest);
+        var resultString = await archiveResult.Content.ReadAsStringAsync();
+
+        var result = JsonSerializer.Deserialize<ArchiveDto>(resultString, new JsonSerializerOptions() { PropertyNameCaseInsensitive = true });
+        var videoItemId = Guid.NewGuid();
+        var updatedArchiveResult = await client.PostAsJsonAsync<Guid>($"/api/Archive/{result?.Id}/VideoItems", videoItemId);
```

Comment:
Doen we het wel nog met /VideoItems? of bij de test op line 90 lijkt het dat je ze ophaalt zonder deze subpath

Reply:
Op lijn 90 worden alle lijsten opgehaald van ieder archief, met /VideoItems haal je alleen alle videoitems van een specifieke archief op.